workitem/link/link_repository_blackbox_test.go

```diff
@@ -0,0 +1,119 @@
+package link_test
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	"github.com/almighty/almighty-core/gormtestsupport"
+	"github.com/almighty/almighty-core/migration"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/space"
+	testsupport "github.com/almighty/almighty-core/test"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/almighty/almighty-core/workitem/link"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type linkRepoBlackBoxTest struct {
+	gormtestsupport.DBTestSuite
+	repo               link.WorkItemLinkRepository
+	clean              func()
+	ctx                context.Context
+	testSpace          uuid.UUID
+	testIdentity       account.Identity
+	TestTreeLinkTypeID uuid.UUID
+}
+
+// SetupSuite overrides the DBTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (s *linkRepoBlackBoxTest) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+	s.ctx = migration.NewMigrationContext(context.Background())
+	s.DBTestSuite.PopulateDBTestSuite(s.ctx)
+}
+
+func TestRunLinkRepoBlackBoxTest(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, &linkRepoBlackBoxTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../../config.yaml")})
+}
+
+func (s *linkRepoBlackBoxTest) SetupTest() {
+	s.repo = link.NewWorkItemLinkRepository(s.DB)
+	s.clean = cleaner.DeleteCreatedEntities(s.DB)
+	testIdentity, err := testsupport.CreateTestIdentity(s.DB, "jdoe1", "test")
+	s.testIdentity = testIdentity
+	require.Nil(s.T(), err)
+
+	// create a space
+	spaceRepository := space.NewRepository(s.DB)
+	spaceName := "test-space" + uuid.NewV4().String()
```

Comment:
fixed it here -> https://github.com/almighty/almighty-core/pull/1235/commits/fcaa9547e11227331d4c4cd9fccb3f39ccbf2c95#diff-bf110de3daad9b25035f4012579bbe55R55